website/database.py

```diff
@@ -608,6 +611,52 @@ def batch_get_adventures(self, adventure_ids):
         keys = {id: {"id": id} for id in adventure_ids}
         return ADVENTURES.batch_get(keys) if keys else {}
 
+    def get_public_adventures_filters(self):
+        filters = {}
+        for record in PUBLIC_ADVENTURES_FILTERS.scan():
+            if record.get("field") == "tag":
+                filters["tag"] = filters.get("tag", [])
+                filters["tag"].append(record.get("value"))
+            else:
+                lang, level = record.get("value").split("#")
+                filters["lang"] = filters.get("lang", [])
+                filters["lang"].append(lang)
+                filters["level"] = filters.get("level", [])
+                filters["level"].append(level)
+        return filters
+
+    def update_public_adventure_filters_indexes(self, adventure):
+        """This function adds/updates filters and indexes that we use to retrieve public adventures."""
+        if not adventure.get("public"):
+            return
+        id = adventure.get("id")
+        date = adventure.get("date")
+        lang_value = adventure.get("language", "en")
+        PUBLIC_ADVENTURES_INDEXES.put({"field_value": f"lang_{lang_value}", "date_adventure_id": f"{date}_{id}"})
+
+        levels = adventure.get("levels")
+        if not levels:
+            levels = [adventure.get("level")]
+        for level in levels:
+            PUBLIC_ADVENTURES_INDEXES.put({"field_value": f"level_{level}", "date_adventure_id": f"{date}_{id}"})
+            value = f"{lang_value}#{level}"
+            PUBLIC_ADVENTURES_FILTERS.put({"field": "lang#level", "value": value})
+        tags = adventure.get("tags", [])
+        for tag in tags:
+            PUBLIC_ADVENTURES_INDEXES.put({"field_value": f"tag#{tag}", "date_adventure_id": f"{date}_{id}"})
+            PUBLIC_ADVENTURES_FILTERS.put({"field": "tag", "value": tag})
+
+    def remove_public_adventure_filters_indexes(self, field, value, adventure):
+        PUBLIC_ADVENTURES_FILTERS.delete({"field": field, "value": value})
+        PUBLIC_ADVENTURES_INDEXES.delete({"field_value": f"{field}_{value}",
+                                          "date_adventure_id": f"{adventure['date']}_{adventure['id']}"})
+
+    def get_public_adventures_indexes(self, key):
+        """This function returns adventure_ids for a specific index.
+        E.g., key={"field_value": "lang_en", "date_adventure_id": "..."}"""
+        result = PUBLIC_ADVENTURES_INDEXES.get_all(key)
```

Comment:
You probably want query them in reverse, so that the newest adventures are first.

Use the Dynamo feature (`reverse=True`).

Reply:
Well you know what, I guess order doesn't matter since you're throwing the results into a `set` afterwards anyway. 

But it could matter, and perhaps should matter, to preserve some sort of ordering and have the newest results on top.